pkg/apis/pipeline/v1beta1/pipeline_validation_test.go

```diff
@@ -1160,14 +1160,32 @@ func TestValidatePipelineResults_Failure(t *testing.T) {
 		results       []PipelineResult
 		expectedError apis.FieldError
 	}{{
-		desc: "invalid pipeline result reference",
+		desc: "invalid pipeline task result reference in parse expression",
+		results: []PipelineResult{{
+			Name:        "my-pipeline-result",
+			Description: "this is my pipeline result",
+			Value:       *NewArrayOrString("$(tasks.myTask.results.resultName.foo.bar)"),
+		}},
+		expectedError: *apis.ErrInvalidValue(`expected all of the expressions [tasks.myTask.results.resultName.foo.bar] to be result expressions but only [] were`, "results[0].value").Also(
+			apis.ErrInvalidValue("referencing a nonexistent task", "results[0].value")),
+	}, {
+		desc: "invalid pipeline task result reference",
 		results: []PipelineResult{{
 			Name:        "my-pipeline-result",
 			Description: "this is my pipeline result",
 			Value:       *NewArrayOrString("$(tasks.a-task.results.output.key1.extra)"),
 		}},
 		expectedError: *apis.ErrInvalidValue(`expected all of the expressions [tasks.a-task.results.output.key1.extra] to be result expressions but only [] were`, "results[0].value").Also(
 			apis.ErrInvalidValue("referencing a nonexistent task", "results[0].value")),
+	}, {
+		desc: "invalid pipeline finally result reference",
```

Comment:
could you please update the description of this test case to describe what is invalid?

Reply:
added variable to the end so that the `desc` still is concise

Reply:
references a finally task that doesnt exist